app/sqaws.py

```diff
@@ -14,27 +21,208 @@ def quote(value):
+    ec2 = boto3.client('ec2', region_name=region_name)
+    print(f'Setting tag {tag_value} on {type_ec2}: {id_name} in region {region_name}')
+    if not dryrun:
+        response = ec2.create_tags(Resources=[id_name], Tags=[{
+            'Key': tag_name,
+            'Value': tag_value
+        }])
+        print(f'Response from create_tags: {str(response)}')
+
+
+# Get 'stop after', 'terminate after', and 'Nagbot state' tag names in a resource, regardless of formatting
+def get_tag_names(tags: dict) -> tuple:
+    stop_after_tag_name, terminate_after_tag_name, nagbot_state_tag_name = 'StopAfter', 'TerminateAfter', 'NagbotState'
+    for key, value in tags.items():
+        if (key.lower()).startswith('stop') and 'after' in (key.lower()):
+            stop_after_tag_name = key
+        if (key.lower()).startswith('terminate') and 'after' in (key.lower()):
+            terminate_after_tag_name = key
+        if (key.lower()).startswith('nagbot') and 'state' in (key.lower()):
+            nagbot_state_tag_name = key
+    return stop_after_tag_name, terminate_after_tag_name, nagbot_state_tag_name
+
+
+# Estimate the monthly cost of an EBS storage volume; pricing estimations based on region us-east-1
+def estimate_monthly_ebs_storage_price(region_name: str, instance_id: str, volume_type: str, size: float, iops: float,
+                                       throughput: float) -> float:
+    if instance_id.startswith('i'):
+        ec2_resource = boto3.resource('ec2', region_name=region_name)
+        total_gb = sum([v.size for v in ec2_resource.Instance(instance_id).volumes.all()])
+        return total_gb * 0.1  # Assume EBS costs $0.1/GB/month when calculating for attached volumes
+
+    if 'gp3' in volume_type:  # gp3 type storage depends on storage, IOPS, and throughput
+        cost = size * 0.08
+        if iops > 3000:
+            provisioned_iops = iops - 3000
+            cost = cost + (provisioned_iops * 0.005)
+        if throughput > 125:
+            provisioned_throughput = throughput - 125
+            cost = cost + (provisioned_throughput * 0.04)
+        return cost
+    else:  # Assume EBS costs $0.1/GB/month, true as of Dec 2021 for gp2 type storage
+        return size * 0.1
+
+
+# Stop an EC2 resource - currently, only instances should be able to be stopped
+def stop_resource(region_name: str, instance_id: str, dryrun: bool) -> bool:
+    print(f'Stopping instance: {str(instance_id)}...')
+    ec2 = boto3.client('ec2', region_name=region_name)
+    try:
+        if not dryrun:
+            response = ec2.stop_instances(InstanceIds=[instance_id])
+            print(f'Response from stop_instances: {str(response)}')
+        return True
+    except Exception as e:
+        print(f'Failure when calling stop_instances: {str(e)}')
+        return False
+
+
+# Class representing a generic EC2 resource & containing functions shared by all resources currently in use
 @dataclass
-class Instance:
+class Resource:
```

Comment:
I'd vote to decompose the new classes into different files such that `sqaws.py` is no more and is replaced with a resource.py, instances.py, and volumes.py